### ISSUE TITLE
Substreams Sink js not working on Nextjs

### ISSUE THREAD

**Issue by harshrjjpt** [substreams-js/substreams-js#55]
I tried using  substreams sink js in my Next.js project. I wanted to create a service to fetch data from any substream dynamically. However, when I used this service in my component, an error occurred: - `error ./node_modules/undici/lib/api/api-connect.js:3:24 Module not found: Can't resolve 'async_hooks'.`

**Comment by fubhy** [comment]
Hi. The sink package is intended for usage with Node applications. It might in theory be possible to make it platform independent but so far that was not a use case I was planning to support. The reason for that is that sinks are generally long running, stateful processes which don't fit well into the serverless execution model.

Can you maybe elaborate what exactly you are trying to accomplish?

**Comment by harshrjjpt** [comment]
I've been working on developing an application to present substream data in a tabular format with filtering capabilities block by block. 
Here is my POC:
[POC](https://twitter.com/harshrjjpt/status/1743912101801025599)

I am using the Substreams Sink js library on the Node.js backend to dynamically fetch substreams data through an API. However, my current objective is to transfer the Node.js functionality to the frontend i.e on NEXT.js. I plan to achieve this by creating a service within the NEXT.js  to retrieve the data.

**Comment by fubhy** [comment]
After checking the code again I don't see why this wouldn't work in Next.js. The sink package is actually free of any node platform specific dependencies. I suspect you are doing sth. within your application that is pulling in Node dependencies. I'll check out your PoC and report back.

**Comment by fubhy** [comment]
Can you create a small reproducible example of what you are trying to accomplish so I can reproduce it locally?

**Comment by fubhy** [comment]
Make sure you are using the correct `@connect/*` web dependency stack. Not the `-node` one.

**Comment by harshrjjpt** [comment]
Hey, thanks for the advice. My mistake – I was utilizing the @connectrpc/connect-node package. I've switched to the correct web dependency stack, and everything is working now.

However, I'm encountering another issue...
when using the GitHub spkg file path. Im getting an error:
 'Access to fetch at 'https://github.com/streamingfast/substreams-uniswap-v3/releases/download/v0.2.8/substreams.spkg' from origin 'http://localhost:3001' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`
 
Interestingly, when I was using the sink library on Node.js, it worked seamlessly with the GitHub files.

**Comment by fubhy** [comment]
That's because you don't have CORS issues in Node.js. CORS is a browser security feature. You'll have to load these via a local proxy for instance.

**Comment by fubhy** [comment]
E.g. look how I'm doing that here also: https://github.com/substreams-js/substreams-js/blob/main/examples/example-web/vite.config.ts

To make this work with next.js I guess you could utilize a serverless / edge function that proxies to github but exposes the file through the local path. Make sure you consider any potential security concerns with that approach though.

**Comment by harshrjjpt** [comment]
Hey! Im using nextjs 13 with App router. Im getting same error after using this approach...

**Comment by fubhy** [comment]
Are you talking about CORS? CORS is a browser security feature and has nothing to do with this library. You will have to solve that I am afraid.
